### ISSUE TITLE
`MethodSignatureMismatch` when extending `ReflectionProperty`

### ISSUE THREAD

**Issue by derrabus** [vimeo/psalm#7357]
`ReflectionProperty::setValue()` and `ReflectionProperty::getValue()` are polymorphic. If you work on a static property you can omit the first parameter which is the object on which you want to access the property.

The following code is valid imho:

```PHP
lass MyReflectionProperty extends ReflectionProperty
{
    public function getValue(?object $object = null): mixed
    {
        return parent::getValue(...func_get_args());
    }

    public function setValue(mixed $objectOrValue, mixed $value = null): void
    {
        parent::setValue(...func_get_args());
    }
}
```

see https://psalm.dev/r/7de1b5f18a and https://3v4l.org/Ie4oo

However, Psalm logs `MethodSignatureMismatch` errors. If I apply the changes suggested by those errors, I end up with code that would exit with a fatal error:

see https://3v4l.org/mX1AF

**Comment by orklah** [comment]
Yeah, that seem correct. Would you mind pushing a PR that changes the callmap on that regard?

Thanks

**Comment by derrabus** [comment]
I gave it a try: #7402

**Comment by weirdan** [comment]
The original snippet no longer results in `MethodSignatureMismatch`, so I think it's solved now.
